src/worker/ingestion/action-manager.ts

```diff
@@ -0,0 +1,63 @@
+import { Action, PluginsServerConfig } from '../../types'
+import { DB } from '../../utils/db/db'
+import { status } from '../../utils/status'
+
+type ActionCache = Record<Action['id'], Action>
+
+export class ActionManager {
+    private ready: boolean
+    private db: DB
+    private actionCache: ActionCache
+
+    constructor(db: DB) {
+        this.ready = false
+        this.db = db
+        this.actionCache = {}
+    }
+
+    public async prepare(): Promise<void> {
+        this.actionCache = await this.db.fetchAllActionsMap()
+        this.ready = true
+    }
+
+    public getAction(id: Action['id']): Action | undefined {
+        if (!this.ready) {
+            throw new Error('ActionManager is not ready! Run actionManager.prepare() before this')
+        }
+        return this.actionCache[id]
+    }
+
+    public async fetchAction(id: Action['id']): Promise<void> {
```

Comment:
The name of this function is misleading. It's more `reloadAction` than "fetch (and return) from the database"

Reply:
Yeah, was definitely on the fence about this, went for `reloadAction` and `dropAction`